src/jit/lsraarmarch.cpp

```diff
@@ -464,9 +464,9 @@ void LinearScan::TreeNodeInfoInitCall(GenTreeCall* call)
             if (!varTypeIsStruct(putArgChild) && !putArgChild->OperIs(GT_FIELD_LIST))
             {
 #ifdef _TARGET_ARM_
-                // The `double` types have been transformed to `long` on arm, while the actual longs
+                // The `double` types could been transformed to `long` on arm, while the actual longs
                 // have been decomposed.
-                const bool isDouble = putArgChild->TypeGet() == TYP_LONG;
+                const bool isDouble = (putArgChild->TypeGet() == TYP_LONG || putArgChild->TypeGet() == TYP_DOUBLE);
```

Comment:
We definitely need to adjust the assert below to handle the TYP_DOUBLE case. However, do we need to set srcCount=2 for TYP_DOUBLE? @CarolEidt ?

Reply:
No, I don't believe we want to set `srcCount=2` for `TYP_DOUBLE`. 
I don't think that `isDouble` provides any value here. I think this would be cleaner:
```
                bool expectedSlots = 1;
#ifdef _TARGET_ARM_
                // The `double` types could have been transformed to `long` on arm, while the actual longs
                // have been decomposed.
                if (putArgChild->TypeGet() == TYP_LONG)
                {
                    argNode->gtLsraInfo.srcCount = 2;
                    expectedSlots = 2;
                }
                else if (putArgChild->TypeGet() == TYP_DOUBLE)
                {
                    expectedSlots = 2;
                }
#endif // !_TARGET_ARM_
                // Validate the slot count for this arg.
                assert(curArgTabEntry->numSlots == expectedSlots);
```

Reply:
I'll fix it.